cice.setup

```diff
@@ -422,6 +420,7 @@ set dorun    = false
 set dosubmit = true
 if (\$?SUITE_BUILD) then
   set dobuild  = "\${SUITE_BUILD}"
+endif
```

Comment:
so from what I understand and tested this forgotten `endif` was the cause of the 

    else: endif not found.

error. I am pretty surprised, since this is inside the heredoc, so it is just some text written to a file, still the shell running `cice.setup` dies with a syntax error...

Reply:
exactly.  this is pretty shocking.  The endif is missing in the job.submit script yet that script never complained.  This is exactly what I had been testing all along.  Then when we run a non-suite with cice.setup, it failed due to the missing endif in the part of the script that was generating the job.submit suite script.  I really cannot believe how this bug was manifested in the end.

Reply:
This kind of weird behavior usually means there's some other bug, doesn't it?  Is it possible to step through the script to find out why it hit that block of code?  Or is this a "compiler" bug?

Reply:
I think this weird behavior is due to the peculiarities of the C shell. As far as I know it is not possible to step a shell script...

So yes I think this is could be classified as a bug in the C shell.